src/modal/modal-window.ts

```diff
@@ -40,6 +41,9 @@ import {ModalDismissReasons} from './modal-dismiss-reasons';
 export class NgbModalWindow implements OnInit,
     AfterViewInit, OnDestroy {
   private _elWithFocus: Element;  // element that is focused prior to modal opening
+  private _preventClose = false;
```

Comment:
This works well if used as locally scoped var:
```ts
this._zone.runOutsideAngular(() => {
      let _preventClose = false;
...
```

Reply:
Good idea, will fix that